client/packages/system/src/Name/ListView/ImportProperties/PropertiesImportModal.tsx

```diff
@@ -0,0 +1,239 @@
+
+  const [bufferedFacilityProperties, setBufferedFacilityProperties] = useState<
+    ImportRow[]
+  >(() => []);
+
+  const importAction = async () => {
+    onChangeTab(Tabs.Import);
+    const numberImportRecords = bufferedFacilityProperties?.length ?? 0;
+    if (bufferedFacilityProperties && numberImportRecords > 0) {
+      const importErrorRows: ImportRow[] = [];
+      // Import count can be quite large, we break this into blocks of 10 to avoid too much concurrency
+      const remainingRecords = bufferedFacilityProperties;
+      while (remainingRecords.length) {
+        await Promise.all(
+          remainingRecords.splice(0, 10).map(async facility => {
+            await mutateAsync(toUpdateNamePropertiesInput(facility))
+              .then(async result => {
+                // Map structured Errors
+                if (result?.__typename === 'Mutations') {
+                  // const errorMessage = mapStructuredErrors(result);
+                  const errorMessage = result;
+                  importErrorRows.push({
+                    ...facility,
+                    errorMessage: errorMessage.__typename,
+                  });
+                  return;
+                }
+              })
+              .catch((err: { message: string }) => {
+                if (!err) {
+                  err = { message: t('messages.unknown-error') };
+                }
+                importErrorRows.push({
+                  ...facility,
+                  errorMessage: err.message,
+                });
+              });
+          })
+        ).then(() => {
+          // Update Progress Bar
+          const percentComplete =
+            10 - (remainingRecords.length / numberImportRecords) * 100.0;
+          setImportProgress(percentComplete);
+          setImportErrorCount(importErrorRows.length);
+        });
+      }
+      if (importErrorRows.length === 0) {
+        const importMessage = t('messages.import-generic', {
+          count: numberImportRecords,
+        });
+        const successSnack = success(importMessage);
+        successSnack();
+        onChangeTab(Tabs.Upload);
+        setBufferedFacilityProperties([]);
+        setErrorMessage('');
+        onClose();
+      } else {
+        // Load the error rows in to the component for review
+        setErrorMessage(t('messages.import-generic'));
+        setBufferedFacilityProperties(importErrorRows);
+        setImportErrorCount(importErrorRows.length);
+        onChangeTab(Tabs.Review);
+      }
+    }
+  };
+
+  const onClickStep = (tabName: string) => {
+    switch (tabName) {
+      case Tabs.Upload:
+        changeTab(tabName as Tabs);
+        break;
+      case Tabs.Review:
+        changeTab(tabName as Tabs);
+        break;
+      case Tabs.Import:
+        // Do nothing, user can't get to the import page without clicking the import button
+        break;
+    }
+  };
+
+  const changeTab = (tabName: Tabs) => {
```

Comment:
<img width="1227" alt="Screenshot 2024-06-25 at 10 09 34 AM" src="https://github.com/msupply-foundation/open-msupply/assets/55115239/3bb0c506-ce67-49f7-91fc-e1c5c58e1c43">

Are you calling this in the right places? The tabs look like this, as though you're on the review tab, the whole way through the process 👀